### DISCUSSION TITLE
Modifying the JWT before it is sent from supabase

### DISCUSSION THREAD

**Discussion by sanketch** [supabase/supabase#3221] [Questions] [answered]
Hey! Is there a way to modify the JWT somehow before it is returned to the user? We want to use CubeJS's multi-tenancy feature, and in order to do that we need to pass in the CubeJS's security context in the token. 

One solution is to send over the supabase token to a backend service that then creates another token with CubeJS. This token is then stored in client local storage. But trying to see if there is an approach where we can modify it before the auth token is sent from supabase to save another trip. Any thoughts?

https://cube.dev/docs/security

**Comment by kangmingtay** [comment, marked as answer]
Hey @sanketch, we expose the `JWT_SECRET` on the dashboard under the "Settings" -> "API" page. You can use that `JWT_SECRET` to sign it after updating the JWT payload. However, do note that this should be done on the backend / server-side instead of on the client because the `JWT_SECRET` should not be exposed.

**Reply by sanketch** [reply]
Hey @kangmingtay thanks for the response!  Not sure if I understand this correctly. Cube has its own secret that they say to sign the token with before it is sent to them to retrieve the correct info. How do I actually update the jwt payload through supabase dashboard to include for example a specific field from the database?
